app/src/js/components/Form/Form.js

```diff
@@ -202,22 +228,23 @@ export class Form extends React.Component {
                 break;
             }
 
-            // retrieve value and errors stored in state
-            let inputId = this.generateComponentId(form.schemaProperty);
+            let inputId = generateComponentId(input.schemaProperty, this.id);
             let { value, error } = inputState[inputId];
+
             // coerce non-null values to string to simplify proptype warnings on numbers
             if (type !== formTypes.list && type !== formTypes.subform && !value && value !== 0) {
               value = String(value);
             }
```

Comment:
Adding this comment here because there's not really a better place for it

Not related to this PR, but the optional tag field makes it really noticeable. Our `button--disabled` class isn't doing what it should. I traced it back to line 25 in `_buttons.scss`

```
  &:disabled {
    opacity: 0.6;
    pointer-events: none;
  }
```

We're using the pseudo selector and not the defined class. 

We could fix this by adding the class as a selector like this:

```
  &:disabled,
  &--disabled {
    opacity: 0.6;
    pointer-events: none;
  }
```

We can also just add this change to a future design ticket, but I think it would make that tag field a little more clear.